pkg/sip/inbound.go

```diff
@@ -1145,12 +1145,31 @@ func (c *sipInbound) swapSrcDst(req *sip.Request) {
 	req.AppendHeader((*sip.FromHeader)(c.to))
 	req.RemoveHeader("To")
 	req.AppendHeader((*sip.ToHeader)(c.from))
+	// Remove all Via headers
+	for req.RemoveHeader("Via") {
+	}
```

Comment:
If I understand correctly, this might be incorrect. Each router is supposed to append it's own Via and then remove that line when requests go back. So I believe we are supposed to add own own line and that's it.

Reply:
The behavior you state above is correct for responses within a given transaction: the responses must have the same VIA headers as the request, and proxies on the way back are supposed to "unpleel" their own Via entries from the response,. The REFER request is however a new request, in a new transaction, sent directly to the server from the INVITE Contact header. As such, only the IP address of the request initiator should be in the Via header. I believe the examples in the REFER RFC confirm the above: https://www.rfc-editor.org/rfc/rfc3515#section-4

Reply:
Ah, you are right, I forgot that's a new request. Makes sense :+1: